### ISSUE TITLE
Please help me with Superbooga prompt format

### ISSUE THREAD

**Issue by shamio** [oobabooga/text-generation-webui#5095]
I would liek to work with Superbooga for giving long inputs and getting responses. I use Notebook tab and after loading data and breaking it into chunks,I am really confused to use the proper format. I would like to implement Superbooga tags (<|begin-user-input|>, <|end-user-input|>, and <|injection-point|>) into the ChatML prompt format. Please tell me how to implemen the Superbooga tags into the ChatML prompt format. The ChatML prompt format is like below:

<|im_start|>system
You are a useful AI assistant. Be as much helpful as you can.
<|im_end|>

<|im_start|>user
Write a rebuttal to the given article.
<|im_end|>

<|im_start|>assistant

**Comment by TheDarkTrumpet** [comment]
I'd really use the 'Discussion' section for asking questions and not opening bugs.  The discussion area is more appropriate for questions.

I don't use the "notebook" tab, but I use the "Default" tab to accomplish much like what you're trying to do.  What you should do is:
1. Go to Parameters -> Instruction Template
2. In the dropdown select 'ChatML'

Pay special attention to the 'User string', 'Bot string', and 'Context'

The main thing you're missing above is the 'Context' portion.

3. Then go to 'Default', and select one of the existing prompts.  Change the sections according to what you need in the ChatML instruction template.
4. Replace the 'Context' portion with your text of the article.

Your prompt may need to be changed to really point out the 'context' portion, to make sure that it is looking in the right spot.

5. In Parameters -> Generation -> max_new_tokens, increase this

Start testing.

One forewarning though.  Articles can get quite long.  You have a n_ctx parameter in models.  This usually sits between 2048 and 4096.  This can be increased, and depending on your architecture, you may not be able to do this adequately.  If you run into memory issues, a much better option may be to use LangChain to summarize the main points of the article then feed it back in to create your rebuttal.  Considerably more work, but context can take up a lot of space.

I'm also providing a template I use for meeting transcripts below.  It follows the Alpaca format, but you can look at the steps above for the Alpaca format, look at my template, and set up yours accordingly.  The `<INPUT>` is where the transcript goes, for my case.  I run between 16 to 32k `n_ctx` with `max_new_tokens` set to 4096.

```
Below is an instruction that describes the meeting summary request, paired with an input that provides the raw transcript of the meeting. Write a response that appropriately completes the request.

### Instruction:
Your goal is to create meeting summary notes.  These are NOT in the form of an email, but intended to be inserted into a document to complement notes taken during the meeting.  DO NOT interpret or use any external information besides what's in the meeting summary.  This is NOT a submission of any kind. Do not worry about who said what in the meeting, but focus on the large scale items.  

In the Summary, provide 1 paragraph of high level discussion topics.  In the "Topics Discussed" section, provide the topic, then a description of the topic.  In the "Action Items" provide a list of potential action items identified.

Please provide 3 sections of information:
#### Summary:
#### Topics Discussed:
#### Action Items:

### Input:
<INPUT>

### Response:
```

**Comment by shamio** [comment]
@ [TheDarkTrumpet](https://github.com/TheDarkTrumpet), thank you so much for your detailed and step-by-step guides. I have some other questions too. 
1. What is the point of using Superbooga and splitting the text into multiple chunks if we want to enter the <input> and not using  <|injection-point|>?
2. In your example, where should i put <|injection-point|>? instead of <INPUT> or where?

**Comment by TheDarkTrumpet** [comment]
> @ [TheDarkTrumpet](https://github.com/TheDarkTrumpet), thank you so much for your detailed and step-by-step guides. I have some other questions too.
> 
> 1. What is the point of using Superbooga and splitting the text into multiple chunks if we want to enter the  and not using  <|injection-point|>?
> 2. In your example, where should i put <|injection-point|>? instead of  or where?

I'm unsure what the injection-point clause is doing.  Personally, I'd omit it and see if you get the results you want before trying to add it.  Sometimes less is more, then fine tune from there.

**Comment by shamio** [comment]
> > @ [TheDarkTrumpet](https://github.com/TheDarkTrumpet), thank you so much for your detailed and step-by-step guides. I have some other questions too.
> > 
> > 1. What is the point of using Superbooga and splitting the text into multiple chunks if we want to enter the  and not using  <|injection-point|>?
> > 2. In your example, where should i put <|injection-point|>? instead of  or where?
> 
> I'm unsure what the injection-point clause is doing. Personally, I'd omit it and see if you get the results you want before trying to add it. Sometimes less is more, then fine tune from there.

Thank you so much for your reply. I believe this method doesn't use the Superbooga and it uses the regular feature of Oobabooga. I am still looking to know how to use Superbooga or SuperboogaV2 :(

**Comment by TheDarkTrumpet** [comment]
Yes, this is mostly the regular features of text-generation-webui, not talking about any extensions.  Much of what I'm talking about is pure instruct-based interworkings.  To simply get your model working in more chat/chat-instruct, go to "Parameters -> Instruction Template" and set it there, then in the "chat" tab things should just work.

If you're looking at Superbooga or SuperboogaV2 (which works best in chat/chat-instruct modes from what I understand of them), and you're using this for role play, then I think you'll find it much nicer going to SillyTavern and using text-generation-webui's API as the backend to that.  SillyTavern also has vector storage inplace for chats/group-chats, as an extension you can enable.

**Comment by shamio** [comment]
Thank you for your suggestion. I am still curious to see how to use Superbooga :)

**Comment by shamio** [comment]
Anyone?

**Comment by shamio** [comment]
?
